lib/commands/debug/index.js

```diff
@@ -0,0 +1,200 @@
+const path = require('path');
+const fs = require('fs');
+const SmapiClient = require('@src/clients/smapi-client');
+const { AbstractCommand } = require('@src/commands/abstract-command');
+const optionModel = require('@src/commands/option-model');
+const AuthorizationController = require('@src/controllers/authorization-controller');
+const ResourcesConfig = require('@src/model/resources-config');
+const CONSTANTS = require('@src/utils/constants');
+const profileHelper = require('@src/utils/profile-helper');
+const jsonView = require('@src/view/json-view');
+const stringUtils = require('@src/utils/string-utils');
+const Messenger = require('@src/view/messenger');
+const CliError = require('@src/exceptions/cli-error');
+const helper = require('./helper');
+const NodejsDebugFlow = require('./nodejs-debug');
+const PythonDebugFlow = require('./python-debug');
+const JavaDebugFlow = require('./java-debug');
+
+const DEBUG_FLOWS = [
+    JavaDebugFlow,
+    NodejsDebugFlow,
+    PythonDebugFlow
+];
+
+class DebugCommand extends AbstractCommand {
+    name() {
+        return 'debug';
+    }
+
+    description() {
+        return 'debug the skill';
+    }
+
+    requiredOptions() {
+        return [];
+    }
+
+    optionalOptions() {
+        return ['debug-port', 'wait-for-attach', 'skill-code-region', 'debug-region', 'profile', 'debug'];
+    }
+
+    handle(cmd, cb) {
+        const debugPort = cmd.debugPort || CONSTANTS.LOCAL_DEBUG.DEFAULT_DEBUG_PORT;
+        const skillCodeRegion = cmd.skillCodeRegion || CONSTANTS.ALEXA.REGION.DEFAULT;
+        const debugRegion = cmd.debugRegion || CONSTANTS.ALEXA.REGION.NA;
+        let skillId, profile;
+
+        try {
+            profile = profileHelper.runtimeProfile(cmd.profile);
+        } catch (error) {
+            Messenger.getInstance().error(error);
+            cb(error);
+        }
+
+        try {
+            new ResourcesConfig(path.join(process.cwd(), CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG));
+            skillId = ResourcesConfig.getInstance().getSkillId(profile);
+        } catch (err) {
+            const error = new CliError('Failed to read project resource file. '
+                + 'Please run the command within a ask-cli project.');
+            Messenger.getInstance().error(error);
+            return cb(error);
+        }
+        if (!stringUtils.isNonBlankString(skillId)) {
+            const error = new CliError(`Failed to obtain skill-id for the given profile - ${profile} from project `
+                + `resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}`);
+            Messenger.getInstance().error(error);
+            return cb(error);
+        }
+
+        this._getAccessTokenForProfile(profile, cmd.debug, (tokenErr, token) => {
```

Comment:
@RonWang should we just use ask-smapi-sdk for new features? Seems like this code can be much shorter(3-5x) if used with ask-smapi-sdk. This way async await can be used.